src/OpenTK/Platform/MacOS/CocoaNativeWindow.cs

```diff
@@ -954,6 +958,22 @@ private void HideBorder()
             suppressResize--;
         }
 
+        // Use when point use window related coordinate system
+        private void MoveCursorInWindow(Point p)
+        {
+            var r = Cocoa.SendRect(
+                windowInfo.Handle,
+                Selector.Get("contentRectForFrameRect:"),
+                Cocoa.SendRect(
+                    windowInfo.Handle,
+                    Selector.Get("frame")));
+
+            var screenPoint = new NSPoint();
+            screenPoint.X = r.X + p.X;
+            screenPoint.Y = r.Y + p.Y;
+            CG.DisplayMoveCursorToPoint(WindowInfo.Handle, screenPoint);
```

Comment:
Loosely related, but `HIDInput`'s `SendPosition` is currently not working. Using this method there (with `IntPtr.Zero` for the display) instead of `CG.WarpMouseCursorPosition(p);` fixes it.

Reply:
I'll be dammed, this actually fixes #668. I'll try and fiddle together a PR for it.

Reply:
Tough it makes you wanna cry on the ideas apple have. If you move your mouse quickly, you get a massive cursor popping up, even with CursorVisible=false.

Reply:
The real solution is likely to run such games full screen (it will also give a pretty huge performance boost). Then you won't see the cursor escaping